kubernetes/resource_kubernetes_pod_v1_test.go

```diff
@@ -76,6 +76,7 @@ func TestAccKubernetesPodV1_basic(t *testing.T) {
 					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.generation"),
 					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.resource_version"),
 					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.uid"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.os.0.name", "linux"),
```

Comment:
I think this could be removed from the basic test since we have a [separated test](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/kubernetes/resource_kubernetes_pod_v1_test.go#L1572) to validate `os`. The OS tests can be extended to validate other fields that depend on OS one, that is the motivation to keep it separated.